packages/catlog/src/stdlib/analyses/reachability/mod.rs

```diff
@@ -0,0 +1,222 @@
+    let init_in_forbbiden = f.iter().any(|v| (0..n_p).all(|p| v[p] <= init[p]));
+    Ok(!init_in_forbbiden)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::dbl::{model::*, theory::*};
+    use crate::stdlib::th_sym_monoidal_category;
+    use crate::zero::name;
+    use std::rc::Rc;
+
+    /// The example petri net has the following structure
+    ///                      t1 t2  t3   
+    /// let i_mat = vec![vec![0, 1, 0],  p1
+    ///                  vec![0, 1, 0],  p2
+    ///                  vec![0, 0, 1]]; p3
+    ///
+    /// let o_mat = vec![vec![1, 0, 0],  p1
+    ///                  vec![0, 0, 1],  p2
+    ///                  vec![0, 1, 0]]; p3
+    ///
+    /// (WARNING: the petri net is drawn incorrectly in Handbook of Model Checking)
+    ///
+    /// Let the forbidden state be (0,0,2).
+    ///
+    /// The algorithm terminates in four steps:
+    /// [(0,0,2)] -> [(0,0,2),(1,1,1)] -> [(0,0,2),(0,1,1),(2,2,0)]
+    /// -> [(0,0,2),(0,1,1),(0,2,0)]
+    /// So the three ways one can reach the forbidden state are:
+    /// 1.) starting in the forbidden state (or any superset)
+    /// 2.) having two tokens in p2
+    /// 3.) having one token in each p2 and p3
+    ///
+    /// Consider using algorithm from "Minimal Coverability Tree Construction
+    /// Made Complete and Efficient" for a more efficient algorithm which allows
+    /// "inf" as a possible specification of an invalid state.
+    #[test]
+    fn validate_reachability() {
+        // Define a petri net
+        let th = Rc::new(th_sym_monoidal_category());
+        let (ob_type, op) = (ModalObType::new(name("Object")), name("tensor"));
+        let mut model = ModalDblModel::new(th);
+        let (p1, p2, p3, t1, t2, t3) =
+            (name("p1"), name("p2"), name("p3"), name("t1"), name("t2"), name("t3"));
+        model.add_ob(p1.clone(), ob_type.clone());
+        model.add_ob(p2.clone(), ob_type.clone());
+        model.add_ob(p3.clone(), ob_type.clone());
+        let [x, y, z] = [p1.clone(), p2.clone(), p3.clone()].map(ModalOb::from);
+        model.add_mor(
+            t1,
+            ModalOb::App(ModalOb::List(List::Symmetric, vec![]).into(), op.clone()),
+            ModalOb::App(ModalOb::List(List::Symmetric, vec![x.clone()]).into(), op.clone()),
+            ModalMorType::Zero(ob_type.clone()),
+        );
+        model.add_mor(
+            t2,
+            ModalOb::App(
+                ModalOb::List(List::Symmetric, vec![x.clone(), y.clone()]).into(),
+                op.clone(),
+            ),
+            ModalOb::App(ModalOb::List(List::Symmetric, vec![z.clone()]).into(), op.clone()),
+            ModalMorType::Zero(ob_type.clone()),
+        );
+        model.add_mor(
+            t3,
+            ModalOb::App(ModalOb::List(List::Symmetric, vec![z.clone()]).into(), op.clone()),
+            ModalOb::App(ModalOb::List(List::Symmetric, vec![y.clone()]).into(), op.clone()),
+            ModalMorType::Zero(ob_type),
+        );
```

Comment:
Note to self (no action needed): we should write a helper function for defining Petri nets, if only to make the tests easier to write. This is painful.

Or better, now that I think of it, once @olynch's #716 we should have textual syntax to define models, which should make writing this sort of test much more convenient.